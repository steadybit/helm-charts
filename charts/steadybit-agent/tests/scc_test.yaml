templates:
  - scc.yaml
  - scc-clusterrole.yaml
  - scc-rolebinding.yaml
chart:
  appVersion: v0.0.0
capabilities:
  majorVersion: 1
  minorVersion: 30
tests:
  - it: not rendering on kubernetes
    asserts:
      - hasDocuments:
          count: 0
  - it: forced rendering on kubernetes
    set:
      securityContextConstraint:
        create: true
        name: "my-scc"
    asserts:
      - matchSnapshot: {}
  - it: rendering by default on openshift
    capabilities:
      apiVersions:
        - "security.openshift.io/v1/SecurityContextConstraints"
    asserts:
      - matchSnapshot: {}
  - it: suppressed on openshift
    capabilities:
      apiVersions:
        - "security.openshift.io/v1/SecurityContextConstraints"
    set:
      securityContextConstraint:
        create: false
    asserts:
      - hasDocuments:
          count: 0
