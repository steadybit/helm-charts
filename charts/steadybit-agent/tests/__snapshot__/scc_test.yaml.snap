forced rendering on kubernetes:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: system:openshift:scc:my-scc
    rules:
      - apiGroups:
          - security.openshift.io
        resourceNames:
          - my-scc
        resources:
          - securitycontextconstraints
        verbs:
          - use
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: system:openshift:scc:my-scc
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: system:openshift:scc:my-scc
    subjects:
      - kind: ServiceAccount
        name: steadybit-agent
        namespace: NAMESPACE
  3: |
    apiVersion: security.openshift.io/v1
    kind: SecurityContextConstraints
    metadata:
      name: my-scc
    priority: null
    runAsUser:
      type: MustRunAs
      uid: 10000
    seLinuxContext:
      type: MustRunAs
rendering by default on openshift:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: system:openshift:scc:steadybit-agent
    rules:
      - apiGroups:
          - security.openshift.io
        resourceNames:
          - steadybit-agent
        resources:
          - securitycontextconstraints
        verbs:
          - use
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: system:openshift:scc:steadybit-agent
      namespace: NAMESPACE
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: system:openshift:scc:steadybit-agent
    subjects:
      - kind: ServiceAccount
        name: steadybit-agent
        namespace: NAMESPACE
  3: |
    apiVersion: security.openshift.io/v1
    kind: SecurityContextConstraints
    metadata:
      name: steadybit-agent
    priority: null
    runAsUser:
      type: MustRunAs
      uid: 10000
    seLinuxContext:
      type: MustRunAs
